source/MaterialXCore/Element.cpp

```diff
@@ -116,7 +116,7 @@ ElementPtr Element::getElementByPath(const std::string& path)
     {
```

Comment:
Nitpick: The elementName could be a const reference to avoid copying the string. The elementNames vector could also be declared const above since it never changes.

Reply:
elementName is still not a reference, the & is missing. Otherwise it looks good, I'm approving now.

Reply:
whoops, will fix.

Reply:
thanks